domain/credential/service/providerservice.go

```diff
@@ -87,3 +83,11 @@ func (s *WatchableProviderService) WatchCredential(ctx context.Context, key core
 	}
 	return s.st.WatchCredential(ctx, s.watcherFactory.NewNotifyWatcher, key)
 }
+
+// InvalidateCredential marks the cloud credential for the given name, cloud, owner as invalid.
+func (s *WatchableProviderService) InvalidateCredential(ctx context.Context, key corecredential.Key, reason string) error {
+	if err := key.Validate(); err != nil {
+		return errors.Errorf("invalid id invalidating cloud credential: %w", err)
```

Comment:
This is horrible. Can we word this any better ?

Reply:
reworded. the existing error message from which this was copied was also changed